internal/k8s/service/resolver.go

```diff
@@ -55,80 +55,48 @@ func NewRefNotPermittedError(inner string) ResolutionError {
 	return ResolutionError{inner, RefNotPermittedErrorType}
 }
 
+func getErrorTypePrefix(errType ServiceResolutionErrorType) string {
```

Comment:
if we wanted to make this more declarative too just to reduce the amount of tracking down helpers and switch/cases, you could move this into something like a `errorTypePrefixMap` that you could also just declare at the beginning of the file.